internal/wrapnix/wrapper.sh.tmpl

```diff
@@ -14,19 +14,21 @@ fi
 {{- end }}
 
 {{/*
-We use ShellEnvHashKey to prevent doing shellenv if the correct environment is
-already set. (perf optimization)
 
 We use the guard to prevent infinite loop if something in shellenv causes 
 another wrapped binary to be called. The guard is specific to this project so shellenv
 could still cause another project's shellenv to be called.
 
 DO_NOT_TRACK=1 can be removed once we optimize segment to queue events.
+
+--omit-wrappers-from-path so that we do not invoke other bin-wrappers from
+this bin-wrapper. Instead, we directly invoke the binary from the nix store, which
+should be in PATH.
 */ -}}
 
-if [[ "${{ .ShellEnvHashKey }}" != "{{ .ShellEnvHash }}" ]] && [[ -z "${{ .ShellEnvHashKey }}_GUARD" ]]; then
+if [[ -z "${{ .ShellEnvHashKey }}_GUARD" ]]; then
```

Comment:
I think the guard can be removed (the guard is in place to avoid infinite wrapper loop but since we're removing it from path it can't happen. 

This is hard to test so we may be better off leaving as is for now. Maybe add a comment

Reply:
yeah, I'll leave a comment. Would rather be super conservative with these changes.